routes/api/contacts.js

```diff
@@ -76,15 +73,33 @@ router.put('/:contactId', async (req, res, next) => {
       email: update.email,
       phone: update.phone,
     });
-    const contacts = await listContacts();
-    if (contacts.some(({ id }) => id === contactId)) {
-      res.json(updateContact(contactId, update, contacts));
-    } else {
+    const contact = await updateContact(contactId, update);
+    if (!contact) {
       res.status(404).json({ message: 'Not found' });
+    } else {
+      res.json(contact);
     }
   } catch (err) {
     res.status(400).json({ message: 'Missing fields' });
   }
 });
 
+router.patch('/:contactId/favorite',async (req, res, next) => {
+  const { contactId } = req.params;
+  const body = req.body;
+  const bodyToValidate = Joi.object({
```

Comment:
Co do joi tutaj i post i put to lepiej abyś wyrzucił tworzenie obiektu (const bodyToValidate = Joi.object()) poza obsługę funkcji (może być na początku pliku lub nawet w innym jak wolisz). Bo teraz za każdym razem przy obsłudze endpointu tworzysz nowy obiekt a powinneś utworzyć go tylko raz i z niego korzystać

Reply:
przesunąłem Joi gdzie indziej.